src/main/kotlin/org/jitsi/jibri/api/xmpp/XmppApi.kt

```diff
@@ -0,0 +1,245 @@
+     * send a new IQ with the subsequent stats after starting the service:
+     * [JibriIq.Status.FAILED] if there was an error starting the service (or an error while the service was running)
+     * [JibriIq.Status.BUSY] if the Jibri was already busy and couldn't start the requested service
+     * [JibriIq.Status.ON] if the service started successfully
+     */
+    private fun handleStartJibriIq(
+        startJibriIq: JibriIq,
+        xmppEnvironment: XmppEnvironmentConfig,
+        mucClient: MucClient
+    ): IQ {
+        logger.info("Received start request")
+        // We don't want to block the response to wait for the service to actually start, so submit a job to
+        // start the service asynchronously and send an IQ with the status after its done.
+        executor.submit {
+            val resultIq = JibriIqHelper.create(startJibriIq.from, IQ.Type.set)
+            try {
+                logger.info("Starting service")
+
+                // If there is an issue with the service while it's running, we need to send an IQ
+                // to notify the caller who invoked the service of its status, so we'll listen
+                // for the service's status while it's running and this method will be invoked
+                // if it changes
+                val serviceStatusHandler: JibriServiceStatusHandler = { serviceStatus ->
+                    when (serviceStatus) {
+                        JibriServiceStatus.ERROR -> {
+                            logger.info("Current service had an error")
+                            val errorIq = JibriIqHelper.create(startJibriIq.from, IQ.Type.set, JibriIq.Status.FAILED)
+                            logger.info("Sending error iq ${errorIq.toXML()}")
+                            mucClient.sendStanza(errorIq)
+                        }
+                        else -> {} // We only care about errors here
+                    }
+                }
+                val startServiceResult = handleStartService(startJibriIq, xmppEnvironment, serviceStatusHandler)
+
+                resultIq.status = when (startServiceResult) {
+                    StartServiceResult.SUCCESS -> JibriIq.Status.ON
+                    StartServiceResult.BUSY -> JibriIq.Status.BUSY
+                    StartServiceResult.ERROR -> JibriIq.Status.FAILED
+                }
+            } catch (e: Throwable) {
+                logger.error("Error in startService task: $e")
+                resultIq.status = JibriIq.Status.FAILED
+            } finally {
+                logger.info("Sending start service response iq: ${resultIq.toXML()}")
+                mucClient.sendStanza(resultIq)
+            }
+        }
+        // Immediately respond that the request is pending
+        val initialResponse = JibriIqHelper.createResult(startJibriIq)
+        initialResponse.status = JibriIq.Status.PENDING
+        logger.info("Sending 'pending' response to start IQ")
+        return initialResponse
```

Comment:
If we add the status as a parameter to `JibriIqHelper.createResult` we can simplify to `return JibriIqHelper.createResult(...)`